apps/web/app/components/hooks/use-user.ts

```diff
@@ -0,0 +1,28 @@
+import { useAccount } from "wagmi";
+import useUserRoleCookie from "./use-user-role-cookie";
+import type { UserRole } from "@/common/user-role-schema";
+import type { Address } from "viem";
+
+function useUser({ required }: { required: false }): { address?: Address; role?: UserRole };
+function useUser(): { address: Address; role: UserRole };
+function useUser({
+  required,
+}: { required?: boolean } = {}):
+  | { address: Address; role: UserRole }
+  | { address?: Address; role?: UserRole } {
```

Comment:
I don't love how this function works. When using something like this I would expect it to return null instead an object with undefined inside.

There could even be 2 hooks, one that returns `{ address: Address; role: UserRole }| null` dependeng on if the users is logged in or not, and another one that returns `{ address: Address; role: UserRole }` or fails.

Reply:
That's exactly how it works! If `required` is false, then all parameters are optional. I agree we could make it or `| null` the entire object instead of each value but it's the same behaviour.